File: Stack/20. Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack <Character> st=new Stack<Character>();
        int flag=0;
        for(int i=0;i<s.length();i++)
        {
            char ch=s.charAt(i);
            if(ch=='('||ch=='{'||ch=='[')
            {
                st.push(ch);
            }
            else
            {
                if(st.isEmpty()==true)
                {
                    flag=1;
                    break;
                }
                char c=st.peek();
                if(ch==')' && c=='(' || ch==']' && c=='[' || ch=='}' && c=='{')
                    st.pop();
                else
                {
                    
                }
            }
        }
        if(st.isEmpty()==true && flag==0)
            return true;
        else
            return false;
    }
}

File: Stack/20. Valid Parentheses.cpp
class Solution
{
public:
    bool isValid(string s)
    {

        stack<char> st;
        for (int i = 0; i < s.size(); i++)
        {
            if (st.empty() && (s[i] == ']' || s[i] == '}' || s[i] == ')'))
                return false;
            else if (s[i] == '(' || s[i] == '{' || s[i] == '[')
                st.push(s[i]);
            else
            {
                char top = st.top();
                if (top == '[' && s[i] == ']')
                {
                    if (!st.empty())
                        st.pop();
                }
                else if (top == '(' && s[i] == ')')
                {
                    if (!st.empty())
                        st.pop();
                }
                else if (top == '{' && s[i] == '}')
                {
                    if (!st.empty())
                        st.pop();
                }
                else
                {
                    return false;
                }
            }
        }
        if (!st.empty())
            return false;

        return true;
    }
};